### PR TITLE
[Doc] add pruning config list doc

### PR DESCRIPTION
**Pull request by J-shang**
### Description ###
add doc about the pruning config list.

### CODE REVIEW

File: docs/en_US/Compression/v2_pruning_config_list.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
+Pruning Config List Interpretation
```
Comment (QuanluZhang) [inline docs/en_US/Compression/v2_pruning_config_list.rst:1]: -> Pruning Config Specification

File: docs/en_US/Compression/v2_pruning_config_list.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+==================================
+
+The Keys in Config List
+-----------------------
+
+Each sub-config in the config list is a dict, and the scope of each setting (key) is only internal to each sub-config.
+If multiple sub-configs are configured with the same module, the later ones will overwrite the previous ones.
```
Comment (QuanluZhang) [inline docs/en_US/Compression/v2_pruning_config_list.rst:8]: with -> for

File: docs/en_US/Compression/v2_pruning_config_list.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+op_types
+^^^^^^^^
+
+The type of the module specified in this sub-config.
```
Comment (QuanluZhang) [inline docs/en_US/Compression/v2_pruning_config_list.rst:13]: specified in -> targeted by

File: docs/en_US/Compression/v2_pruning_config_list.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+If ``op_names`` is not set in this sub-config, all modules in the model that satisfy the type will be selected.
+If ``op_names`` is set in this sub-config, the selected module should satisfy both type and name.
+
+op_names
+^^^^^^^^
+
+The name of the module specified in this sub-config.
```
Comment (QuanluZhang) [inline docs/en_US/Compression/v2_pruning_config_list.rst:20]: module specified in -> modules targeted by

File: docs/en_US/Compression/v2_pruning_config_list.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+If ``op_types`` is set in this sub-config, the selected module should satisfy both type and name.
+
+op_partial_names
+^^^^^^^^^^^^^^^^
+
+This key is for the modules to be pruned with similar names. NNI will find all names in the model,
```
Comment (QuanluZhang) [inline docs/en_US/Compression/v2_pruning_config_list.rst:26]: here "similar" means matching prefix, or matching sub-string, or others?

File: docs/en_US/Compression/v2_pruning_config_list.rst [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+find names that contain one of ``op_partial_names``, and append them into the ``op_names``.
+
+sparsity
+^^^^^^^^
+
+The sparsity ratio of each selected module.
```
Comment (QuanluZhang) [inline docs/en_US/Compression/v2_pruning_config_list.rst:32]: For a module, we prune its weight by default. How to specify output pruning? Is this a valid use case?

File: docs/en_US/Compression/v2_pruning_config_list.rst [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+e.g., the ``sparsity`` is 0.8 means each selected module will mask 80% values on the weight.
+If ``layer_1`` (500 parameters) and ``layer_2`` (1000 parameters) are selected in this sub-config,
+then ``layer_1`` will be masked 400 parameters and ``layer_2`` will be masked 800 parameters.
+
+sparsity_per_layer
+^^^^^^^^^^^^^^^^^^
+
+Another name for ``sparsity``.
```
Comment (QuanluZhang) [inline docs/en_US/Compression/v2_pruning_config_list.rst:41]: which one do you recommend? "sparsity" or "sparsity_per_layer"?

File: docs/en_US/Compression/v2_pruning_config_list.rst [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+e.g., the ``total_sparsity`` is 0.8 means 80% of parameters in this sub-config will be masked.
+If ``layer_1`` (500 parameters) and ``layer_2`` (1000 parameters) are selected in this sub-config,
+then ``layer_1`` and ``layer_2`` will be masked a total of 1200 parameters,
+how these total parameters are distributed between the two layers is determined by the pruning algorithm.
+
+max_sparsity_per_layer
+^^^^^^^^^^^^^^^^^^^^^^
+
+This key is usually used with ``total_sparsity``. It limits the maximum sparsity ratio of each layer.
```
Comment (QuanluZhang) [inline docs/en_US/Compression/v2_pruning_config_list.rst:56]: "layer" and "module" have the same meaning?

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 6 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst:32]: We don't support pruning output yet.

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 8 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst:56]: yes, they are the same, I will unify them to `layer`

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 7 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst:41]: I think `sparsity_per_layer` is better in v2.

In v1, `sparsity` has both meaning `sparsity_per_layer` and `total_sparsity`, and in most cases, it means `sparsity_per_layer`. So for taking care of v1 users, we define `sparsity` same as `sparsity_per_layer`.

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 5 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst:26]: means `matching same sub-string`, will fix it

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 1 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst]: fixed

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 2 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst]: fixed

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 3 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst]: fixed

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 4 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst]: fixed

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 5 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst]: fixed

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 8 above)
Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst:56]: fixed

File: docs/en_US/Compression/v2_pruning_config_list.rst (hunk 7 above)
Comment (QuanluZhang) [reply docs/en_US/Compression/v2_pruning_config_list.rst:41]: so you recommend `sparsity_per_layer`, then we can make it clear. you can move the description from `sparsity` to `sparsity_per_layer`, then under `sparsity` you can say it has the same meaning as `sparsity_per_layer` but will be deprecated in future.

Comment (J-shang) [reply docs/en_US/Compression/v2_pruning_config_list.rst]: That's a good suggestion, fixed it

**Review by QuanluZhang — APPROVED** [review]

File: docs/en_US/Compression/v2_pruning_config_list.rst [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+In ``total_sparsity`` example, there are 1200 parameters that need to be masked and all parameters in ``layer_1`` may be totally masked.
+To avoid this situation, ``max_sparsity_per_layer`` can be set as 0.9, this means up to 450 parameters can be masked in ``layer_1``,
+and 900 parameters can be masked in ``layer_2``.
+
+exclude
+^^^^^^^
+
+This key cannot exist in a sub-config at the same time with keys containing ``sparsity``.
```
Comment (zheng-ningxin) [inline docs/en_US/Compression/v2_pruning_config_list.rst:66]: The `exclude` and `sparsity` keyword are mutually exclusive and cannot exist in the same sub-config.

**Review by zheng-ningxin — APPROVED** [review]

**Pull request merged**